presto-spi/src/main/java/io/prestosql/spi/type/Timestamps.java

```diff
@@ -22,10 +22,11 @@
 import static java.lang.Math.floorMod;
 import static java.lang.String.format;
 
-class Timestamps
+public class Timestamps
```

Comment:
This class seems to nearly be a susbet of `DateTimes`. Could the rest of its functionality be moved here (and then this class could also be renamed `DateTimes`)?

If that's too much to move, the two classes should at least define the same set of conversion constants and functions, and have comments referring to each other (and a todo to merge them).

Reply:
Yes, agreed. There is some functionality that is internal to the engine that should not be exposed to connectors, hence it's not in the SPI. We also try to keep the SPI as "implementation-free" as possible, but some things are needed for the basic type implementation.

Once we're done adding `time(p)` and `time(p) with time zone`, we should do some cleanup. Some of the things that make sense when I added `timestamp(p)` stopped making sense as I rolled in support for other types.